js_modules/dagster-ui/packages/ui-core/src/pipelines/PipelineOrJobDisambiguationRoot.tsx

```diff
@@ -13,8 +16,17 @@ export const PipelineOrJobDisambiguationRoot = (props: Props) => {
   const location = useLocation();
   const {pipelinePath} = useParams<{pipelinePath: string}>();
 
-  const {pipelineName: pipelineOrJobName} = explorerPathFromString(pipelinePath);
+  const {loading} = useContext(WorkspaceContext);
+  const {loading: permissionsLoading} = useContext(PermissionsContext);
   const repo = useRepository(repoAddress);
+
+  useBlockTraceUntilTrue('Workspace', loading);
+  useBlockTraceUntilTrue('Permissions', permissionsLoading);
+  if (loading || permissionsLoading) {
```

Comment:
Since this site uses `useRepository` it's also secretly depending on the RootWorkspaceQuery being loaded.